pageserver/src/disk_usage_eviction_task.rs

```diff
@@ -477,9 +488,15 @@ pub(crate) async fn disk_usage_eviction_task_iteration_impl<U: Usage>(
             if let Some(next) = next {
                 match next {
                     Ok(Ok(file_size)) => {
+                        METRICS.layers_evicted.inc();
                         usage_assumed.add_available_bytes(file_size);
                     }
-                    Ok(Err((file_size, EvictionError::NotFound | EvictionError::Downloaded))) => {
+                    Ok(Err((
+                        file_size,
+                        EvictionError::NotFound
+                        | EvictionError::Downloaded
+                        | EvictionError::Timeout,
+                    ))) => {
                         evictions_failed.file_sizes += file_size;
                         evictions_failed.count += 1;
```

Comment:
No metric for the error case because we already have metrics inside `layer.rs` ?

Reply:
No, I did not add that. But we do count the started, completed and cancelled evictions.

An eviction can be cancelled if you have a read after you've wanted to evict something which is not yet deleted from disk (read wins).

Timeout will not cause a cancelled eviction. Like I wanted to do originally: have just `evict` instead of `evict_and_wait` but I failed to see a way to fix the tests.

Reply:
Ok just wanted to point out that it seems like we're not counting the errors explicitly. 

> No, I did not add that. But we do count the started, completed and cancelled evictions.

Are you saying if we need error count we can get it by `error_count = started - completed`

Good enough for me, might be worth a block comment in the `metrics.rs` , but maybe it's obvious.

Reply:
> Are you saying if we need error count we can get it by error_count = started - completed

No, it's `hanging_count = start - completed - sum by (instance) (cancelled)`. As I've been monitoring the data, it's quite difficult to setup a warning or alert because it can legitimately hang for multiple samples, then some other eviction could hang for 1..n samples so the count stays non-zero.

Reply:
Also block comment not added -- block comments are bad anyways, and the dashboard has been using the correct calculation for N months.